src/data/set/intervals/instances.lean

```diff
@@ -0,0 +1,104 @@
+/-
+Copyright (c) 2022 Stuart Presnell. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Stuart Presnell, Eric Wieser, Yaël Dillies
+-/
+import data.set.intervals.basic
+import algebra.order.ring
+import algebra.group_power.order
+
+/-!
+# Structure instances for unit intervals
+
+For suitably structured underlying type `α`, we exhibit the structure of
+the unit intervals from `0` to `1`.
+
+Note: Instances for the interval `Ici 0` are dealt with in `algebra/order/nonneg.lean`.
+
+## TODO
+* structure instances for intervals -1 to 1
+* structure instances for `Ici 1`
+* structure instances for `(Ioo (-1) 1)ᶜ`
+* `group_with_zero (Icc 0 1)` for `linear_ordered_field α`
```

Comment:
```suggestion
```
I was being stupid!

Reply:
No problem! You had me puzzling on that for a little while :)